### ISSUE TITLE
Blank Canvas: Add additional "Link in Bio" block patterns

### ISSUE THREAD

**Issue by kjellr** [Automattic/themes#3161]
A handful of additional block patterns have been designed for Blank Canvas. We'll use this issue to track their development. 

- [x] **Emoji Links**
- [x] **Gradient Links**
- [x] **Musician Links**
- [x] **Product Links**
- [x] **Text Links**
- [ ] **Jetpack Subscribe + Links** / 🚫 Blocked by https://github.com/Automattic/jetpack/issues/19113
- [ ] **PayPal Button + Links**  / 🚫 Blocked by https://github.com/Automattic/jetpack/issues/18684
- [ ] **Podcast Episode** / 🚫 Blocked by https://github.com/WordPress/gutenberg/issues/24231
- [ ] **Image Links** / 🚫 Blocked by https://github.com/WordPress/gutenberg/issues/10057

## Screenshots:

Emoji Links|Gradient Links
---|---
![emoji-links](https://user-images.githubusercontent.com/1202812/106647785-640efd80-655d-11eb-8f1d-91f70536da47.jpg)|![Untitled](https://user-images.githubusercontent.com/1202812/106648067-c536d100-655d-11eb-8be8-c8891512d34b.jpg)

Musician Links|Product Links
---|---
![Untitled](https://user-images.githubusercontent.com/1202812/106648700-9ff69280-655e-11eb-8f7b-4c8f69bc11ec.jpg)|![Untitled-2](https://user-images.githubusercontent.com/1202812/106648699-9f5dfc00-655e-11eb-8d82-f891094a5b8c.jpg)

Text Links|Jetpack Subscribe + Links
---|---
![Untitled](https://user-images.githubusercontent.com/1202812/106648892-e0eea700-655e-11eb-8ad0-d71c782606d7.jpg)|![subscribe](https://user-images.githubusercontent.com/1202812/106648949-f237b380-655e-11eb-9e60-4614c5d92609.jpg)

PayPal Button + Links|Podcast Episode
---|---
<img width="1136" alt="Screen Shot 2021-02-02 at 2 11 47 PM" src="https://user-images.githubusercontent.com/1202812/106650200-9ec66500-6560-11eb-872c-f379a39d658c.png">* This one still needs some design iteration.|![podcast](https://user-images.githubusercontent.com/1202812/106650071-750d3e00-6560-11eb-8f1e-a36d7d370409.jpg)

Image Links|&nbsp;
---|---
![link-in-bio-patterns](https://user-images.githubusercontent.com/1202812/106650317-c61d3200-6560-11eb-80a4-2f32d2ab6b0a.jpg)|![link-in-bio-patterns](https://user-images.githubusercontent.com/1202812/106650399-db925c00-6560-11eb-9f09-4f9350849a5a.jpg)

**Comment by simison** [comment]
@ianstewart @ollierozdarz @kjellr would it be possible to ship at least some of these to all customers? I think they're great and design picker grid is lacking this kind of simple designs.

**Comment by alaczek** [comment]
Sure thing! We'll go over those with @ianstewart and I'll get on adding them to the pattern site.

**Comment by alaczek** [comment]
When will those be available on .com (I mean in the theme)?

Edit: NVM, I see they are under a separate category "Link in bio", not under "Blank Canvas" (which is slightly confusing because they do come from Blank Canvas theme).

I might be overthinking this, but if we ever create a "Link in bio" category for our stand-alone patterns, would that cause conflict with a pattern category registered by the theme?

**Comment by kjellr** [comment]
> Edit: NVM, I see they are under a separate category "Link in bio", not under "Blank Canvas" (which is slightly confusing because they do come from Blank Canvas theme).

We plan to spin these patterns out into their own separate "Link in Bio" theme in the showcase, so that's why they're separated out for now. It'll be essentially a duplicate of Blank Canvas, but with one of those patterns defined as the homepage by default so that users can easily see it and switch to it. 

> I might be overthinking this, but if we ever create a "Link in bio" category for our stand-alone patterns, would that cause conflict with a pattern category registered by the theme?

I haven't tested this, but I _think_ they'll just be merged into the same category (If they don't, we should file a bug about that).

**Comment by simison** [comment]
> We plan to spin these patterns out into their own separate "Link in Bio" theme in the showcase, so that's why they're separated out for now. It'll be essentially a duplicate of Blank Canvas, but with one of those patterns defined as the homepage by default so that users can easily see it and switch to it.

Does it have to be a separate theme, or should it be just same theme surfaced twice, with different template?

**Comment by kjellr** [comment]
TBD — @jeffikus and team are sorting out the solution.

**Comment by ianstewart** [comment]
> I might be overthinking this, but if we ever create a "Link in bio" category for our stand-alone patterns, would that cause conflict with a pattern category registered by the theme?
— Ola

> I haven't tested this, but I think they'll just be merged into the same category (If they don't, we should file a bug about that).
— Kjell

+1 to this. cc @andrewserong. We should expect themes to be able to register any number of categories that may or may not duplicate any number of categories with our WPcom patterns. So our patterns toolkit should be able to adapt to that. And, in the editor, perhaps merge these categories. I think I might expect theme patterns with the same category name to appear as if they're prepended to the list of WPcom categories under that same name in the editor.

**Comment by kjellr** [comment]
It looks like the **Jetpack Subscribe + Links** pattern may be unblocked now. I'll take a look at implementing that soon.

_Edit: The fix in the linked issue still needs to be deployed to WP.com — will try again next week._

**Comment by andrewserong** [comment]
> We should expect themes to be able to register any number of categories that may or may not duplicate any number of categories with our WPcom patterns

Thanks for the ping @ianstewart! Yes, when we register patterns and categories pulled from the patterns toolkit, the categories get merged. Pattern registration matches against a category slug, so theme patterns and dotcompatterns patterns with the same category will all get included together. I'm not 100% sure of the sort order of patterns within the category, it'd likely depend on which pattern registration function runs first (the ETK plugin or the theme), but we can probably tweak that with action priority if we want to swap it.

**Comment by alaczek** [comment]
We started porting some of the block patterns from Blank Canvas to our pattern source site so that they are available to everyone. BUT won't that mean that on sites running Blank Canvas, we'll get duplicated patterns in the same category (for example Link in Bio)?

**Comment by kjellr** [comment]
Each pattern has its own unique slug — I wonder if we can filter out duplicates with that?

**Comment by alaczek** [comment]
Maybe... I'm not sure what makes a slug for the universal patterns we're creating. If it's the post slug from the source site it might be tricky to match the theme as we can't have duplicates. I'd love to hear from people with more knowledge of the inner workings tho :)

**Comment by kjellr** [comment]
Just noting that the **Subscription Form and Links** pattern was almost ready to go, but during testing we uncovered another Jetpack bug that made it rough to use (https://github.com/Automattic/jetpack/issues/19113). The pattern has been temporarily disabled in #3445, and I've unchecked it in the issue above.
